lib/debounce.js

```diff
@@ -1,16 +1,11 @@
-import requestAnimationFrame from './requestAnimationFrame'
+export default function debounce (func, time = 300) {
+  let timeoutId = null
 
-export default function debounce (fn) {
-  let waiting = false
-
-  return function (...args) {
-    if (!waiting) {
-      waiting = true
-
-      requestAnimationFrame(() => {
-        fn(...args)
-        waiting = false
-      })
+  return function () {
```

Comment:
Hmm, yeah, I didn't think about the fact that the height of the collapsible is transitioned. My thinking behind coupling it with the paint is that if your application is running slowly, you're automatically going to execute the costly update only when it is necessary (right before painting), but still never end up painting a frame where the height is incorrect. But if the height is transitioned anyway, those in-between frames don't really matter.

If the window is resized 10 times over a period of 500ms, the height of the collapsible will _never_ be resized until the window has finished resizing, as the debounce will delay the initial execution by 300ms, and then keep delaying it every 50ms, which means that the collapsible won't be resized until after 800ms (+ however long it is until the next paint after the timeout finally executes).

I'm not crazy about that delay that will be introduced. If you're resizing the page, there will be a noticable delay until the collapsible starts transitioning. Could you maybe play with the timeout value so that the update is never executed until the resizing is done, but the delay is low enough that it's not so noticable?


Then again, if the height of the collapsible is transitioned anyway, those in-between frames might not matter, since